lib/messages.rb

```diff
@@ -27,7 +27,7 @@ def validation_message(errors)
     puts errors
```

Comment:
Here is a concrete dependency on `puts`.  It would be nice to see a `Logger` class which could have various levels of logging and the ability to pass in a different output. 

It would be nice to have something like:

```ruby
class Logger 
  def initialize(args = {})
    @output = lambda { |message|  args[:output](message) }
  end

  def self.game(message)
    @output.call(message)
  end

  def self.info(message)
    @output.call("[Info] " + msg)
  end
end
```

Then be able to call:
```ruby
class Game
  def initialize(args = {})
    @logger = Logger.new(puts)
  end

  logger.game(Messages.prompt_guess)
end
```

Reply:
Very cool. Is "Logger" a common name for something like this? Is "wrapper" common?

I would think it'd be confusing to have it be Logger if it's not logging errors and messages, given Ruby's built-in https://ruby-doc.org/stdlib-2.4.0/libdoc/logger/rdoc/Logger.html.

Reply:
Creating a logger which is specific to the application is pretty common.  

This class is using dependency injection, we are injecting the dependent (`puts`) function which will take the output from the program and do something with it.  There is no longer a concrete dependency on `puts`.

In ruby, of course we do not want to override the built in logger. In most cases we would call this  something like `GameLogger`.

If you don't want to have the logging and display output in the same class, you could call this class something like `Display` or `UserInterface`.

Also I updated the Logger class above to actually work, it did not work as written previously.

Reply:
Cool, I'll work on this for the next iteration. Thanks!